lib/internal/Magento/Framework/GraphQl/Query/ErrorHandler.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Framework\GraphQl\Query;
+
+/**
+ * Class ErrorHandler
+ *
+ * @package Magento\Framework\GraphQl\Query
+ */
+class ErrorHandler implements ErrorHandlerInterface
+{
+    /**
+     * @var \Magento\Framework\Logger\Monolog
+     */
+    private $clientLogger;
+
+    /**
+     * @var \Magento\Framework\Logger\Monolog
+     */
+    private $serverLogger;
+
+    /**
+     * @var array
+     */
+    private $clientErrorCategories;
+
+    /**
+     * @var array
+     */
+    private $serverErrorCategories;
+
+    /**
+     * @var \Magento\Framework\Logger\Monolog
+     */
+    private $generalLogger;
+
+    /**
+     * ErrorHandler constructor.
+     *
+     * @param \Magento\Framework\Logger\Monolog $clientLogger
+     * @param \Magento\Framework\Logger\Monolog $serverLogger
+     * @param \Magento\Framework\Logger\Monolog $generalLogger
+     * @param array                             $clientErrorCategories
+     * @param array                             $serverErrorCategories
+     *
+     * @SuppressWarnings(PHPMD.LongVariable)
+     */
+    public function __construct(
+        \Magento\Framework\Logger\Monolog $clientLogger,
+        \Magento\Framework\Logger\Monolog $serverLogger,
+        \Magento\Framework\Logger\Monolog $generalLogger,
+        array $clientErrorCategories = [],
+        array $serverErrorCategories = []
+    ) {
+        $this->clientLogger = $clientLogger;
+        $this->serverLogger = $serverLogger;
+        $this->generalLogger = $generalLogger;
+        $this->clientErrorCategories = $clientErrorCategories;
+        $this->serverErrorCategories = $serverErrorCategories;
+    }
+
+    /**
+     * Handle errors
+     *
+     * @param \GraphQL\Error\Error[] $errors
+     * @param callable               $formatter
+     *
+     * @return array
+     */
+    public function handle(array $errors, callable $formatter):array
+    {
+        return array_map(
+            function (\GraphQL\Error\ClientAware $error) use ($formatter) {
+                $this->logError($error);
+
+                return $formatter($error);
+            },
+            $errors
+        );
+    }
+
+    /**
+     * @param \GraphQL\Error\ClientAware $error
+     *
+     * @return boolean
+     */
+    private function logError(\GraphQL\Error\ClientAware $error):bool
+    {
+        if (in_array($error->getCategory(), $this->clientErrorCategories)) {
```

Comment:
@paliarush 
Maybe will be better to take a decision based on the type of exception but not on `category`?

Like all `Localized` exception are related to `client errors`

Reply:
@naydav 
I would suggest using`\GraphQL\Error\ClientAware::isClientSafe` for making this decision.
Reasons:
1. logic of errors displaying is the same as errors logging
2. no need to be tied to exception classes - so no need for additional di configs

Reply:
by the way, in case of `\GraphQL\Error\ClientAware::isClientSafe` :
1. we get rid of General logger (isClientSafe returns bool only)
2. we get rid of Exception classes list in di.xml